lib/eslint.js

```diff
@@ -891,8 +874,20 @@ module.exports = (function() {
 
     // copy over methods
     Object.keys(externalMethods).forEach(function(methodName) {
+        var exMethodName = externalMethods[methodName];
         api[methodName] = function() {
-            return sourceCode ? sourceCode[externalMethods[methodName]].apply(sourceCode, arguments) : null;
+            if (!sourceCode) {
+                return null;
+            }
+
+            // Avoid `Function.prototype.apply` because of slow.
+            switch (arguments.length) {
```

Comment:
Yeah, so, try this:

```js
api[methodName] = function(a, b, c, d, e) {
   // ...

   return sourceCode[exMethodName](a, b, c , d, e)
};
```

I don't think we have any methods with more than five arguments.